sound/soc/sof/core.c

```diff
@@ -104,6 +104,8 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 		return ret;
```

Comment:
The commit message could be improved a bit. I think you are adding "a state machine" and not "the state machine". :) I'd reword the start as: "Add an explicit state machine to track stages of firmware boot. This reduce number of distinct state tracking variables and is helpful e.g. when determining the actions to perform at system suspend".